src/dataflow/src/source/mod.rs

```diff
@@ -849,6 +853,221 @@ enum MessageProcessing {
     YieldedWithDelay,
 }
 
+type EventSender =
+    mpsc::Sender<Event<Option<Timestamp>, Result<(Row, Timestamp, Diff), SourceError>>>;
+
+/// An active transaction at a particular point in time. An instance of this struct is provided to
+/// a source when calling start_tx() on its timestamper. This has the effect of freezing the
+/// timestamper clock while the data for the transaction is sent.
+///
+/// The transaction is automatically committed on Drop at which point the timestamper will continue
+/// ticking its internal clock.
+pub struct Transaction<'a> {
```

Comment:
Can we use some other terminology please, like `SourceTransaction`, `RealTimeSourceTransaction`, or similar?

There are other notions of transaction in the system (for example, the stuff Matt is working on which is not so closely related to sources), and I think it will be confusing if we have something called just plain `Transaction` modeling the transactions in an upstream DB.

Reply:
Makes sense, I switched to `SourceTransaction`